src/celeritas/optical/WavelengthShiftParams.hh

```diff
@@ -0,0 +1,58 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/WavelengthShiftParams.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Types.hh"
+#include "corecel/data/CollectionMirror.hh"
+#include "corecel/data/ParamsDataInterface.hh"
+#include "celeritas/io/ImportOpticalMaterial.hh"
+
+#include "WavelengthShiftData.hh"
+
+namespace celeritas
+{
+struct ImportData;
+
+namespace optical
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Build and manage wavelength shift (WLS) data.
+ */
+class WavelengthShiftParams final
```

Comment:
Yes, this can be consolidated into a WLS model when it is added later. For now, I just want to add an interactor as the first step until we feature out how to support multiple WLS models/processes in a organized way.

Reply:
I might misunderstand, but these `Params` refer to data needed by the interactor rather than the MFP grid, similar to something like `em/KleinNishinaData.hh`, correct? I think WLS is unique in that it's the only one of the interactors that needs on-device data, whereas Rayleigh's data wasn't used by the interactor and purely for building MFPs.

Reply:
Ah, gotcha. That makes sense.

Reply:
@whokion what were your thoughts on how to support multiple WLS models? Could we e.g. as a start just have multiple instances of a single `WavelengthShiftModel` with different WLS properties as inputs (and I suppose some way to determine which MFPs went with which model)?

Reply:
@amandalund One tricky part is how to support multiple models/cases for different ranges of WLS absorption lengths (ABS) and output spectra within a single process.  There are several different cases: 1)  input optical photon ranges (either in energy or wavelength) of ABS and output spectra are completely exclusive each other, which is the simpliest case that can be handled by one model and one interactor with a combined data, 2)  part of ABS ranges are overlapped, but with completely exclusive output spectra  which is the case that multiple models can be compete for the WLS process (based on their relative absorption probabilities) and produce output spectrum with the corresponding data and in the target range if the selected WLS model wins over other discrete processes 3) both of ABS ranges and spectra are overlapped (i,e., output of WLS may be input of WLS2, which may be treated as a similar way as 2) with some variations, 4) one ABS input range, but have two different output spectra with different time structures (two transition levels from one WLS material).  Apparently, Geant4 tries to support these different cases with the combination of two separated/independent WLS and WLS2 with different input data (so, users must know exactly what they want to simulate and how to configure different cases correctly).   At this moment, I do not have a clear picture that is different from the Geant4 approach, so add this initial interactor with a minimal data structure as a starting point.  Of course, other pieces of good ideas are welcomed for the better design.

Reply:
So since we *do* have an optical model class, I think it would be good to update that before merging.

Reply:
Actually we need to add a separate WLS2 if we want to follow what exactly Geant4 does.  Otherwise, we need to combine different cases into one convolved model.  So, if you do not mind, let's leave implementation options for the next MR.

Reply:
I don't see why we can't do exactly what Geant4 does with a single WLS model class.  Couldn't we do something like:
```c++
auto wls1 = std::make_shared<WavelengthShiftModel>(action_id++, imported, ImportModelClass::wls1);
auto wls2 = std::make_shared<WavelengthShiftModel>(action_id++, imported, ImportModelClass::wls2);
```
where the last argument determines which imported WLS optical properties and MFPs to use? The two models will compete for the discrete interaction like in Geant4.

Reply:
Yes, this is equivalent to what Geant4 does (assuming that there is an additional mechnism to call a common interactor with different wls data to generate WLS photon with a different/corresponding spectrum depending on which one is selected).  The main point is whether we can combine these two identical models in the step competition into single one with a combined data (which is more general, I think) - it is straightforward if ranges of ABS are exclusive each other as I mentioned earlier, but not clear how to do it cleanly for other cases even though it is a purely technical question.

Reply:
Yes, launching the interactors should work fine without any doing anything special since the models use exactly the same interactor and data classes. I think this would be the simplest way to do things for now, and we could think about having a "combined" model in the future.

Reply:
Yes, I agree.  Chatting with @hhollenb, he also has a decent idea how to combine them multiple WLS processes for the future improvement.